symfony/templates/order/header.html.twig

```diff
@@ -82,11 +82,20 @@
             </div>
         {% endif %}
     {% endif %}
-    {% if activeStep == 'finalized' and showUnfinalizeMsg and finalizeForm.vars.valid %}
-        <div class="alert alert-danger well-sm">
-            <i class="fa fa-info-circle" aria-hidden="true"></i>
-            <strong>Order saved but not finalized. Please finalize the order.</strong>
-        </div>
+    {% if activeStep == 'finalized' %}
+        {% if siteInfo.siteId != order.site %}
+            <div class="alert alert-danger well-sm">
+                <i class="fa fa-info-circle" aria-hidden="true"></i>
+                <strong>Warning! The site you are logged into {{ app.session.get('currentSiteDisplayName') }} does not match the site of order creation {{ siteInfo.siteDisplayName(order.site) }}. If you logged into a different site by error, please switch to the correct site.</strong>
```

Comment:
As seen in the screenshots, this seems to run together a bit when you read it. I think adding a `[]` around the site name will help legibility. May need to consult with the product team to confirm.

```suggestion
                <strong>Warning! The site you are logged into [{{ app.session.get('currentSiteDisplayName') }}] does not match the site of order creation [{{ siteInfo.siteDisplayName(order.site) }}]. If you logged into a different site by error, please switch to the correct site.</strong>
```